templates/observability-remote-write-proxy.yml

```diff
@@ -0,0 +1,337 @@
+              mountPath: ${OBSERVABILITY_REMOTE_WRITE_PROXY_OIDC_CONFIG_FILEDIR}
+          volumes:
+           - name: observability-remote-write-proxy-oidc-config
+             secret:
+               secretName: observability-remote-write-proxy-oidc-config
+               defaultMode: 288 # 0440 in octal mode permissions
+               optional: true
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      labels:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+      name: observability-rw-proxy
+    spec:
+      ports:
+        - name: http
+          port: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PORT}}
+          targetPort: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PORT}}
+          protocol: TCP
+      selector:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      labels:
+        app.kubernetes.io/component: prometheus-remote-write-proxy-metrics
+        app.kubernetes.io/name: observability-remote-write-proxy-metrics
+      name: observability-rw-proxy-metrics
+    spec:
+      ports:
+        - name: metrics
+          port: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_METRICS_PORT}}
+          targetPort: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_METRICS_PORT}}
+          protocol: TCP
+      selector:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+  - kind: PodDisruptionBudget
+    apiVersion: policy/v1
+    metadata:
+      name: observability-remote-write-proxy-pdb
+    spec:
+      minAvailable: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PDB_MIN_AVAILABLE}}
+      selector:
+        matchLabels:
+          app.kubernetes.io/name: observability-remote-write-proxy
+  - kind: NetworkPolicy
+    apiVersion: networking.k8s.io/v1
+    metadata:
+      labels:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+      name: observability-remote-write-proxy
+    spec:
+      podSelector:
+        matchLabels:
+          app.kubernetes.io/component: prometheus-remote-write-proxy
+          app.kubernetes.io/name: observability-remote-write-proxy
+      policyTypes:
+      - Ingress
+      ingress:
+      - from:
+          # Allow access from the default OpenShift Ingress Controller pods.
+          # The default OpenShift Ingress Controller pods are located in a K8s
+          # Namespace that contains the 'network.openshift.io/policy-group' K8s
+          # label key with the value 'ingress' and the 'name' label with the
+          # value 'openshift-ingress'. Additionally, those pods have the label
+          # 'ingresscontroller.operator.openshift.io/deployment-ingresscontroller'
+          # with the value 'default'.
+        - namespaceSelector:
```

Comment:
This assumes that in stage / prod cluster there is a namespace called openshift-ingress with the label `network.openshift.io/policy-group: ingress`. It also assumes that there are some pods in that namespace with the label `ingresscontroller.operator.openshift.io/deployment-ingresscontroller: default`.
This is the case with clusters created from OCM, but it might be different in the case of AppSRE clusters. In case we detect any issue this should be readapted if we decide to keep the NetworkPolicy